app/src/org/odk/collect/android/activities/FormHierarchyActivity.java

```diff
@@ -320,6 +325,62 @@ public void refreshView() {
         FormEntryActivity.mFormController.jumpToIndex(currentIndex);
     }
 
+    public int getFormEntryPromptIcon(FormEntryPrompt fep){
+        if (BuildConfig.DEBUG) {
+            Log.i("FEPICON", "FEP (" + fep.hashCode() + ") data type is: " + fep.getDataType() + " | control type is: " + fep.getControlType());
+        }
+        switch(fep.getControlType()){
+            case Constants.CONTROL_SELECT_ONE:
+                return R.drawable.avatar_vellum_single_answer;
+            case Constants.CONTROL_SELECT_MULTI:
+                return R.drawable.avatar_vellum_multi_answer;
+            case Constants.CONTROL_TEXTAREA:
+                return R.drawable.avatar_vellum_text;
+            case Constants.CONTROL_SECRET:
+                return R.drawable.avatar_vellum_password;
+            case Constants.CONTROL_LABEL:
+                return R.drawable.avatar_vellum_label;
+            case Constants.CONTROL_AUDIO_CAPTURE:
+                return R.drawable.avatar_vellum_audio_capture;
+            case Constants.CONTROL_VIDEO_CAPTURE:
+                return R.drawable.avatar_vellum_video;
+            case Constants.CONTROL_TRIGGER:
+                return R.drawable.avatar_vellum_question_list;
+            case Constants.CONTROL_IMAGE_CHOOSE:
+                return R.drawable.avatar_search;
+            case Constants.CONTROL_RANGE:
+            case Constants.CONTROL_UPLOAD:
+            case Constants.CONTROL_SUBMIT:
+            case Constants.CONTROL_INPUT:
+                return getDrawableIDFor(fep);
+        }
+        return -1;
+    }
+
+    public static int getDrawableIDFor(FormEntryPrompt fep){
```

Comment:
My personal preference is that adding the type of the parameter to the method name is unnecessary in a statically-typed language.